agent/agent_test.go

```diff
@@ -2729,6 +2730,37 @@ tasks:
 	s.Equal(expectedLines, actualLines)
 }
 
+func (s *AgentSuite) TestAttemptsToClearGitConfig() {
+	projYml := `
+tasks:
+- name: this_is_a_task_name
+  commands:
+  - command: shell.exec
+    params:
+      script: |
+        echo hello
+`
+	s.setupRunTask(projYml)
+
+	nextTask := &apimodels.NextTaskResponse{
+		TaskId:     s.tc.task.ID,
+		TaskSecret: s.tc.task.Secret,
+	}
+	_, _, err := s.a.runTask(s.ctx, s.tc, nextTask, false, s.testTmpDirName)
```

Comment:
The test would be a little more comprehensive if it created a `.gitconfig` file in advance (or made it in the task command) and later verified that it was deleted.

Reply:
I had that set up but had to remove it because it's hard to write it in a way that doesn't clear the actual git config of the user running it locally.

Reply:
It seems like it shouldn't be using the real home dir since it already sets the test's temp dir as the home directory. Or if we're really concerned about accidentally modifying the home dir in this test, we could just `require` and fail the test early if we find that the task's home dir is either unset or set to `$HOME`.

Reply:
Correct. What was hard was creating a task that does something like `git config --global user.name "Your Name"` and then checking if it was cleared because the git config would use the actual home directory. I mocked it more though and just manually create the file.